src/main/java/org/jenkinsci/plugins/docker/workflow/client/DockerClient.java

```diff
@@ -285,4 +316,29 @@ public ContainerRecord getContainerRecord(@Nonnull EnvVars launchEnv, String con
                 (created != null ? created.getTime() : 0L), 
                 Collections.<String,String>emptyMap());
     }
+
+    /**
+     * Inspect the volumes of a docker image/container.
+     * @param launchEnv Docker client launch environment.
+     * @param objectId The image/container ID.
+     * @return a set of volumes of the docker image/container.
+     * @throws IOException Execution error. Also fails if cannot retrieve the requested field from the request
+     * @throws InterruptedException Interrupted
+     */
+    public Set<String> getVolumes(@Nonnull EnvVars launchEnv, String objectId) throws IOException, InterruptedException {
+        LaunchResult result = launch(launchEnv, true, "inspect", objectId);
+        if (result.getStatus() != 0) {
+            return Collections.EMPTY_SET;
+        }
+
+        JsonParser parser = new JsonParser();
+        JsonElement information = parser.parse(result.getOut());
+        JsonObject volumes = information.getAsJsonArray().get(0).getAsJsonObject().getAsJsonObject("Config").getAsJsonObject("Volumes");
```

Comment:
I think you could avoid the GSON dep by just calling

```java
inspect(launchEnv, objectId, "{{.Config.Volumes}}")
```

or something along these lines.